### PR TITLE
Introduce jump (new formula)

### PR DESCRIPTION
**Pull request by gsamokovarov**
[Jump] is an [autojump] and [z] alternative that comes with fuzzy
matching and easy shell integrations for bash, zsh and fish.

[Jump]: https://github.com/gsamokovarov/jump
[autojump]: https://github.com/wting/autojump
[z]: https://github.com/rupa/z

### CODE REVIEW

File: Library/Formula/jump.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    mkdir_p buildpath/"src/github.com/gsamokovarov"
+    ln_s buildpath, buildpath/"src/github.com/gsamokovarov/jump"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go build -o jump"
```
Comment (bfontaine) [inline Library/Formula/jump.rb]: Use `system "go", "build", "-o", "#{bin}/jump"` instead and remove `bin.install "jump"` below. Thanks!

File: Library/Formula/jump.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    bin.install "jump"
+    man1.install "man/jump.1"
+  end
+
+  def caveats; <<-EOS.undent
+    To install shell integration, follow the instructions at:
+      #{prefix}/bin/jump shell
+    EOS
+  end
```
Comment (bfontaine) [inline Library/Formula/jump.rb]: If these caveats are not Homebrew-specific we can remove them. Also, `#{bin}` is equivalent to `#{prefix}/bin`.

File: Library/Formula/jump.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  test do
+    mkdir_p testpath/"test_dir"
+    ENV["JUMP_HOME"] = testpath.to_s
+    system "jump chdir #{testpath/"test_dir"}"
+
+    assert_equal testpath/"test_dir", shell_output("jump cd tdir").chomp
```
Comment (bfontaine) [inline Library/Formula/jump.rb]: Please use `#{bin}/jump` instead of `jump`.

File: Library/Formula/jump.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  def caveats; <<-EOS.undent
+    To install shell integration, follow the instructions at:
+      #{prefix}/bin/jump shell
+    EOS
+  end
+
+  test do
+    mkdir_p testpath/"test_dir"
+    ENV["JUMP_HOME"] = testpath.to_s
+    system "jump chdir #{testpath/"test_dir"}"
```
Comment (bfontaine) [inline Library/Formula/jump.rb]: Please use one string per argument: `system "jump", "chdir", "#{testpath}/test_dir"

File: Library/Formula/jump.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  end
+
+  def caveats; <<-EOS.undent
+    To install shell integration, follow the instructions at:
+      #{prefix}/bin/jump shell
+    EOS
+  end
+
+  test do
+    mkdir_p testpath/"test_dir"
```
Comment (bfontaine) [inline Library/Formula/jump.rb]: You can use `(testpath/"test_dir").mkpath` for that.

File: Library/Formula/jump.rb (hunk 2 above)
Comment (gsamokovarov) [inline Library/Formula/jump.rb]: Thanks for the hint! I think we should keep this line (or some form of it) as you gotta integrate it with your shell to make it usable. E.g. get the fancy `j` shell function.

**Comment by gsamokovarov** [comment]
Whoa @bfontaine, that was fast! Thanks for the swift review!

File: Library/Formula/jump.rb [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    mkdir_p buildpath/"src/github.com/gsamokovarov"
+    ln_s buildpath, buildpath/"src/github.com/gsamokovarov/jump"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "-o", "jump"
+    bin.install "jump"
```
Comment (bfontaine) [inline Library/Formula/jump.rb]: Use `"-o", "#{bin}/jump"` so you can remove `bin.install "jump"` :wink:

**Comment by gsamokovarov** [comment]
Fixed the build. All the checks are :green_heart: now!

File: Library/Formula/jump.rb [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+    system "go", "build", "-o", "#{bin}/jump"
+    man1.install "man/jump.1"
+  end
+
+  def caveats; <<-EOS.undent
+    To install the shell integration, follow the instructions at:
+      #{bin}/jump shell
+    EOS
+  end
```
Comment (bfontaine) [inline Library/Formula/jump.rb]: @mikemcquaid, thoughts on keeping/removing the caveats here?

Comment (mikemcquaid) [inline Library/Formula/jump.rb]: I think unless they are Homebrew-specific instructions (which they aren't): better to just assume people will look at the homepage for this. If the "integration" is e.g. bash completion it'd be ideal to just install that as usual for people.

Comment (gsamokovarov) [inline Library/Formula/jump.rb]: The main interface of the tool is a shell function called `j`, which comes from the shell integrations. Without it, you won't get the same experience.

Comment (mikemcquaid) [inline Library/Formula/jump.rb]: Unless that applies just to Homebrew: let's skip the caveats.

Comment (gsamokovarov) [inline Library/Formula/jump.rb]: Omitted the caveats in the latest amend.

**Comment by mikemcquaid** [comment]
Thanks @gsamokovarov!

**Comment by gsamokovarov** [comment]
:confetti_ball: :star2:

**Pull request closed without merging**